llvm/lib/ProfileData/MemProf.cpp

```diff
@@ -53,6 +53,7 @@ IndexedMemProfRecord::deserialize(const MemProfSchema &Schema,
           endian::readNext<FrameId, llvm::endianness::little, unaligned>(Ptr);
       Node.CallStack.push_back(Id);
     }
+    Node.CSId = hashCallStack(Node.CallStack);
```

Comment:
Will this eventually be replaced by the CSId held in the indexed profile?

Reply:
Yes.  As we transition from the current version to the next, we will most likely have two deserializers, say `IndexedMemProfRecord::deserializeOld` and `IndexedMemProfRecord::deserializeNew`.  The new one will obtain `CSId` directly from the disk.